### ISSUE TITLE
Crash SDWebImageDownloaderOperation.m line 405

### ISSUE THREAD

**Issue by cherpake** [rs/SDWebImage#1947]
# OS Version: 10.2.1 (14D27)
# Device: iPad 4
# RAM Free: 35.7%
# Disk Free: 9%

#12. Crashed: NSOperationQueue 0x15594c70 :: NSOperation 0x155a1de0 (QOS: DEFAULT)
0  libobjc.A.dylib                0x1d1044e6 objc_msgSend + 5
1  SDWebImage                     0x4d2543 -[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] (SDWebImageDownloaderOperation.m:405)
2  SDWebImage                     0x4d0715 -[SDWebImageDownloader URLSession:task:didCompleteWithError:] (SDWebImageDownloader.m:307)
3  CFNetwork                      0x1e52fef7 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.185 + 76
4  Foundation                     0x1e7c05f1 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 8
5  Foundation                     0x1e71300f -[NSBlockOperation main] + 146
6  Foundation                     0x1e70416b -[__NSOperationInternal _start:] + 774
7  Foundation                     0x1e7c2873 __NSOQSchedule_f + 190
8  libdispatch.dylib              0x1d546e5d _dispatch_queue_serial_drain + 840
9  libdispatch.dylib              0x1d53d379 _dispatch_queue_invoke + 566
10 libdispatch.dylib              0x1d548a5d _dispatch_root_queue_drain + 378
11 libdispatch.dylib              0x1d548883 _dispatch_worker_thread3 + 106
12 libsystem_pthread.dylib        0x1d6ef937 _pthread_wqthread + 1168
13 libsystem_pthread.dylib        0x1d6ef490 start_wqthread + 8

### New Issue Checklist

[*] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
[*] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
[*] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 10.2.1
 SDWebImage Version      |  3.8.2
 Integration Method      |  cocoapods 
 Xcode Version           |  Xcode 8.3
 Repro rate              | only once
 Repro with our demo prj | no
 Demo project link       | none

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by mythodeia** [comment]
i will close this one as there are multiple for the same reason. please use the search. thanks

**Comment by cherpake** [comment]
I wasn't able to find any :(
Sorry

**Comment by mythodeia** [comment]
https://github.com/rs/SDWebImage/issues/1716
https://github.com/rs/SDWebImage/issues/1807
https://github.com/rs/SDWebImage/issues/1816
https://github.com/rs/SDWebImage/issues/1821
https://github.com/rs/SDWebImage/issues/1859
https://github.com/rs/SDWebImage/issues/1883
all of them related to crashes on SDWebImageDownloaderOperation on v4

**Comment by cherpake** [comment]
Yes, they do - but SDWebImageDownloaderOperation is a class and can have multiple crash scenarios unrelated one to another, and most of those are talking about cancelInternally, mine crash is in the successful completion of the NSURLSession task. Slightly different - but I'm probably missing something here.

**Comment by zdl51go** [comment]
[#1940](https://github.com/rs/SDWebImage/pull/1940)   didn't fix this...
